docs/K8sGateKeeper.mdx

```diff
@@ -0,0 +1,378 @@
+
+Admission webhooks in K8s are HTTP callbacks that receive 'admission requests' and do something with them. In particular, K8s-gatekeeper is a special type of admission webhoook: 'ValidatingAdmissionWebhook', which can decide whether to accept or reject this admission request or not. As for admission requests, they are HTTP requests describing an operation on specified resources of K8s (for example, creating/deleting a deployment). For more about admission webhooks, see K8s official doc <https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks>
+
+### 1.2 An example illustrating how it works.
+
+
+For example, when somebody wants to create a deployment containing a pod running nginx (using kubectl or k8s clients), K8s will generate an admission request, which (if translated into yaml format) can be something like this.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.1
+        ports:
+        - containerPort: 80
+```
+
+This request will go through the process of all the middleware shown in the picture, including our K8s-gatekeeper. K8s-gatekeeper can detected all the Casbin enforcers stored in K8s's etcd, which is created and maintained by user(via kubectl or go-client we provide). Each enforcer contains a Casbin model and a Casbin policy. The admission request will be processed by every enforcer, one by one, and only by passing all enforcers can a request be accepted by this K8s-gatekeeper.
+
+(If you do not understand what is Casbin enforcer, model or policy, see this document see: [Get Started](/docs/get-started))
+
+For example, for some reason, the administrator want to forbid the apperance of image 'nginx:1.14.1' while allowing  'nginx:1.3.1', an enforcer containing the following rule and policy can be created: (We will explain how to create an enforcer, what these models and policies and how to write them in following chapters.)
+
+model:
+```ini
+[request_definition]
+r =  obj
+
+[policy_definition]
+p =  obj,eft
+
+[policy_effect]
+e = !some(where (p.eft == deny))
+
+[matchers]
+m = r.obj.Request.Namespace == "default" && r.obj.Request.Resource.Resource =="deployments" && \
+access(r.obj.Request.Object.Object.Spec.Template.Spec.Containers , 0, "Image") == p.obj
+
+```
+
+policy: 
+```csv
+p, "nginx:1.13.1",allow
+p, "nginx:1.14.1",deny
+```
+
+By creating an enforcer containg model and policy above, the previous admission request will be reject by this enforcer, which means K8s won't create this deployment.
+
+
+## 2 Install K8s-gatekeeper
+
+Three methods are provided for installing K8s-gatekeeper: External webhook, Internal webhook and helm.
+
+:::note title="Note"
```

Comment:
It's all my fault.
refer to https://docusaurus.io/docs/markdown-features/admonitions
The proper format should be:
```mdx
:::note title(optional)

content

:::
```

`title` is optional.
It is necessary to leave two blank lines. becase: https://github.com/casbin/casbin-website-v2/pull/41

In addition to `note`, you can also use `tip`, `info`, `caution`, `danger`.
![image](https://user-images.githubusercontent.com/53544726/187452275-4072bd2c-f026-4bcb-8ec9-d5fbcfb264bc.png)